### ISSUE TITLE
ns.flags default boolean true cannot be changed by command line

### ISSUE THREAD

**Issue by dpedroz** [bitburner-official/bitburner-src#725]
### Steps to reproduce:

1. create the script `flags.js` with argsSchema like following;
```
const argsSchema = [
  ['weaken', true],
  ['grow', false]  
];

export async function main(ns) {
  const cfg = ns.flags(argsSchema);
  ns.tprint(JSON.stringify(cfg));
}
``` 

3. run command from terminal
`run flags.js --weaken false --grow true`

ACTUAL: 
`{ "weaken":true, "grow":true }`

SHOULD BE: 
`{ "weaken":false, "grow":true }`

**Comment by rayanth** [comment]
with some further testing, it appears to be an issue specifically related to `false` being passed into args.

Tested: 
Simplified the script and made it more similar to the [ns.flags](https://github.com/bitburner-official/bitburner-src/blob/dev/markdown/bitburner.ns.flags.md) documentation example: 

```
export async function main(ns) {
  const data = ns.flags([
    ['weaken', true],
  ]);
  ns.tprint(data);
}
```
With no args: 
```
[home /]> run flags.js
Running script with 1 thread(s), pid 17 and args: [].
flags.js: {"_":[],"weaken":true}
```
With args: 
```
[home /]> run flags.js --weaken false
Running script with 1 thread(s), pid 18 and args: ["--weaken",false].
flags.js: {"_":[false],"weaken":true}
```
Reversing the code to default false: 
```
export async function main(ns) {
  const data = ns.flags([
    ['weaken', false],
  ]);
  ns.tprint(data);
}
```
With no args: 
```
[home /]> run flags.js
Running script with 1 thread(s), pid 20 and args: [].
flags.js: {"_":[],"weaken":false}
```
With args: 
```
[home /]> run flags.js --weaken true
Running script with 1 thread(s), pid 21 and args: ["--weaken",true].
flags.js: {"_":[true],"weaken":true}
```

So, revealed by testing, it's fine with a `false` as default, and lets me change it to `true`. But if it's default `true`, I cannot change it to `false`.

**Comment by Snarling** [comment]
Boolean flags do not use a second value, they should be false if not provided and true if provided. That's why `run flags.js --weaken true` still passes true as part of the non-flags args, because --weaken is a boolean arg and therefore does not consume the next argument, it is just set to true if the flag is present.

This is not specific to Bitburner, it's just how boolean command line flags work in general. If you want to reverse the logic on the flag, change the flag to be something like `noWeaken`. Alternately, if you're using flags for weaken/hack/grow, but you want it to use weaken if none of the flags were set, then you just do that logic inside the program after the flags processing.

**Comment by elidoran** [comment]
Definitely, the key is in that example the `--weaken` is a separate arg from the `false` which makes sense for command line argument parsing. rayanth's example showing the `false` in the unused args gives it away (the underscore key means unused or unknown args). 

Some cli's support a false value with the prefix `no-`, like this: `--no-weaken`, or providing an explicit value like `--weaken=false` with some conversions for true/false to boolean. Depends on who's making the argument parser.

For your issue trying to make the weaken flag mean dont weaken I'd do the below which uses false as the default but flips the value into its own variable:

```js
const
  flags = ns.flags([ ['weaken', false], ['grow', false] ])
, weaken = ! flags.weaken // flip
, grow   = flags.grow
;
```

For my scripts, I like them to support being called by me with flags, and by other scripts via run or exec with an arg specifying it. (Sometimes I think it'd be nice if we could specify flags in the run/exec functions). So, I do something like this:

```js
const
  flags = ns.flags([ ['stock', false], ['print', false] ])
, stock = flags.stock || (ns.args[0] || '').includes('stock')
, print = flags.print || (ns.args[0] || '').includes('print')
;
```

The includes() also helps when I have multiple options because I can specify them as a single argument by separating them with a comma and the includes still finds the option. Like:

```js
// these are all the same:
ns.exec('g.js', 's0', 123, 'ecorp', 'stock,print');
run g.js ecorp --stock --print
run g.js ecorp stock,print
```
Anyway, I imagine that was some frustrating moments wondering why it wouldn't "just work".  I hope this info helps.

**Comment by Snarling** [comment]
@elidoran
> (Sometimes I think it'd be nice if we could specify flags in the run/exec functions).

You can use flags from ns.run or ns.exec, e.g. `ns.run("test.js", 1, "--booleanFlag", "--numberFlag", 1);` would set the flags booleanFlag to true and numberFlag to 1.

**Comment by elidoran** [comment]
@Snarling  Thank you for pointing it out.  After I posted that message I thought about it and realized it should be obvious. A little embarassing, but, at least now I know. I then went around and tweaked my scripts. I think I was thinking of passing an object with flags that match the script's flag spec into run() and exec(), but, should have realized that'd be a no-no. 

Side note, I also looked into the args stuff and found there's some cool features I wasn't aware of:

1. '--tail' is supported by the game. I'd been using a '--notail' option and coding in running `ns.tail()` when that wasn't specified. 
2. there's support for player defined [autocomplete](https://github.com/bitburner-official/bitburner-src/blob/v2.4.0/src/Terminal/getTabCompletionPossibilities.ts#L331) for the scripts by exporting an autocomplete function from the script. So helpful. Great feature. I'd been testing args count and printing my own usage message to remind me which args exist for that script. Making an autocompleter is way better. 
3. when the flags spec doesn't contain a second arg the terminal will require a value after its use in the terminal (it shows an error if not provided). Def should have been obvious, too, but, I use flags with defaults all the time.
